zcash_client_backend/proto/compact_formats.proto

```diff
@@ -33,16 +41,30 @@ message CompactTx {
     //    valueBalance + (sum(vPubNew) - sum(vPubOld) - sum(tOut))
     uint32 fee = 3;
 
-    repeated CompactSpend spends = 4;
-    repeated CompactOutput outputs = 5;
+    repeated CompactSaplingSpend spends = 4;
+    repeated CompactSaplingOutput outputs = 5;
+    repeated CompactOrchardAction actions = 6;
+}
+
+// CompactSaplingSpend is a Sapling Spend Description as described in 7.3 of the Zcash
+// protocol specification.
+message CompactSaplingSpend {
+    bytes nf = 1;   // nullifier (see the Zcash protocol specification)
 }
 
-message CompactSpend {
-    bytes nf = 1;
+// output is a Sapling Output Description as described in section 7.4 of the
+// Zcash protocol spec. Total size is 948.
```

Comment:
The size of an OutputDescriptionV4 as encoded in a v4 transaction is 948 bytes. The size of an OutputDescriptionV5 as encoded in a v5 transaction is 756 bytes. The size of a CompactSaplingOutput as encoded here is 116 bytes. The latter does not include `cv`, `outCiphertext`, or `zkproof` fields, as well as truncating the `encCiphertext` field.

Also, there is an important semantic distinction between "epk" and `ephemeralKey` in the spec: `ephemeralKey` is a sequence of 32 bytes (which prior to activation of ZIP 214 with Canopy could be a non-canonical encoding), whereas epk refers to the abstract public key. Please consider renaming this field (I'm not including that in the suggestion because I know it requires other Go code changes).

```suggestion
// output encodes the `cmu` field, `ephemeralKey` field (here inaccurately called
// `epk`), and a 52-byte prefix of the `encCiphertext` field of a Sapling Output
// Description. These fields are described in section 7.4 of the Zcash protocol spec.
// Total size is 116 bytes.
```

Reply:
> Please consider renaming this field (I'm not including that in the suggestion because I know it requires other Go code changes).

We can easily do this, because fields are identified by their index, not their name.